src/Modules/SetName/Controllers/SetNameController.cs

```diff
@@ -0,0 +1,32 @@
+﻿using EvoSC.Common.Controllers.Attributes;
+using EvoSC.Manialinks;
+using EvoSC.Modules.Official.SetName.Interfaces;
+using EvoSC.Modules.Official.SetName.Models;
+
+namespace EvoSC.Modules.Official.SetName.Controllers;
+
+[Controller]
+public class SetNameController : ManialinkController
+{
+    private readonly ISetNameService _setNameService;
+
+    public SetNameController(ISetNameService setNameService) => _setNameService = setNameService;
+    
+    public async Task EditNameAsync(SetNameEntryModel input)
+    {
+        if (!IsModelValid)
+        {
+            await ShowAsync(Context.Player, "SetName.EditName", new {input.Nickname});
+            return;
+        }
+
+        await _setNameService.SetNicknameAsync(Context.Player, input.Nickname);
+        await HideAsync("SetName.EditName");
```

Comment:
Doesn't this hide the manialink for all people. There might be multiple people changing their nickname at a time.